src/EditorFeatures/Core.Wpf/IWpfTextViewExtensions.cs

```diff
@@ -1,18 +1,20 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.VisualStudio.Text.Editor;
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
 {
     internal static class IWpfTextViewExtensions
     {
-        public static void SizeToFit(this IWpfTextView view)
+        public static void SizeToFit(this IWpfTextView view, IThreadingContext threadingContext)
         {
             void firstLayout(object sender, TextViewLayoutChangedEventArgs args)
             {
-                view.VisualElement.Dispatcher.BeginInvoke(new Action(() =>
+                threadingContext.JoinableTaskFactory.RunAsync(async () =>
                 {
+                    await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.